clippy_lints/src/methods/mod.rs

```diff
@@ -4339,6 +4340,34 @@ declare_clippy_lint! {
     "using `NonZero::new_unchecked()` in a `const` context"
 }
 
+declare_clippy_lint! {
+    /// ### What it does
+    /// Checks for string slices immediantly followed by `as_bytes`.
+    ///
+    /// ### Why is this bad?
+    /// It involves doing an unnecessary UTF-8 alignment check which is less efficient, and can cause a panic.
+    ///
+    /// ### Known problems
+    /// In some cases, the UTF-8 validation and potential panic from string slicing may be required for
+    /// the code's correctness. If you need to ensure the slice boundaries fall on valid UTF-8 character
+    /// boundaries, the original form (`s[1..5].as_bytes()`) should be preferred.
+    ///
+    /// ### Example
+    /// ```rust
+    /// let s = "Lorem ipsum";
+    /// s[1..5].as_bytes();
+    /// ```
+    /// Use instead:
+    /// ```rust
+    /// let s = "Lorem ipsum";
+    /// &s.as_bytes()[1..5];
+    /// ```
+     #[clippy::version = "1.72.0"]
+     pub SLICE_AS_BYTES,
+     pedantic,
```

Comment:
nit: performance?

Reply:
https://github.com/rust-lang/rust-clippy/pull/10984#discussion_r1267335309
I decided to follow @Jarcho 's category selection
But without a doubt I can revert it back to `perf`

Reply:
Yeah, I think the behavior change should not matter. It is indeed a very minor performance win so that's a somewhat compelling argument for pedanticness. Curious what @Jarcho thinks now?